### PR TITLE
Fixing a bunch of findbugs warnings

### PR DESCRIPTION
**Pull request by sahilTakiar**

### CODE REVIEW

File: gobblin-api/src/main/java/gobblin/configuration/State.java [hunk 1]
```diff
@@ -297,4 +296,9 @@ public void write(DataOutput out) throws IOException {
   public String toString() {
     return properties.toString();
   }
+
+  @Override
+  public int hashCode() {
+    return id == null ? this.properties.hashCode() : this.properties.hashCode() + this.id.hashCode();
```
Comment (chavdar) [inline gobblin-api/src/main/java/gobblin/configuration/State.java]: I suggest using the auto-generated hashCode() implementation from Eclipse.

Comment (chavdar) [inline gobblin-api/src/main/java/gobblin/configuration/State.java]: Please have a look at http://stackoverflow.com/questions/113511/best-implementation-for-hashcode-method for a reasonably good implementation of hashCode().

Comment (sahilTakiar) [inline gobblin-api/src/main/java/gobblin/configuration/State.java]: Sounds good. I updated the code to use the eclipse generated hashCode functions

**Comment by chavdar** [comment]
LGTM

**Comment by sahilTakiar** [comment]
Merged

**Pull request merged**